### PR TITLE
Setup code linting by pre-commit

### PR DESCRIPTION
**Pull request by gi-wg2**
This sets up automatic linting using pre-commit hooks

Note that this requires coursier to be installed.
This is included in .tool-versions for those running `rtx`.

### CODE REVIEW

**Review by alain-wg2 — APPROVED** [review]

File: java/.mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -13,105 +13,107 @@
  * See the License for the specific language governing permissions and
```
Comment (norrs) [inline java/.mvn/wrapper/MavenWrapperDownloader.java]: This is upstream diff for the wrapper thing.

File: java/README.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## Install
 
 ### Maven
+
```
Comment (norrs) [inline java/README.md]: Yes! This is done by a markdown linter I suppose. This is correct.

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+name: pre-commit
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths-ignore:
+      - '*.md'
```
Comment (norrs) [inline .github/workflows/pre-commit.yml]: Why?

File: .github/workflows/pre-commit.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - '.gitignore'
+  pull_request:
+    paths-ignore:
+      - '*.md'
```
Comment (norrs) [inline .github/workflows/pre-commit.yml]: ditto

File: .tool-versions [hunk 5]
```diff
@@ -0,0 +1,2 @@
+coursier   latest
```
Comment (norrs) [inline .tool-versions:1]: coursier is?

File: .github/workflows/pre-commit.yml (hunk 3 above)
Comment (gi-wg2) [reply .github/workflows/pre-commit.yml]: Random copy/paste from the existing action we had

File: .tool-versions (hunk 5 above)
Comment (gi-wg2) [reply .tool-versions:1]: Something used by the pre-commit hook... I would ideally want the pre-commit hooks to be independent though

**Pull request merged**